### ISSUE TITLE
[FEATURE_REQUEST] Consider BookBrainz as a future data provider

### ISSUE THREAD

**Issue by felysu** [mateusz-bak/openreads-android#90]
BookBrainz is a new user-based book database with a clean interface and an active team, which could be very useful as an alternative data provider in the app.
For example, it'd be nice if the user could choose between searching on OpenLibrary, BookBrainz, or both.

Bookbrainz website: https://bookbrainz.org/

BookBrainz is part of a bigger foundation called MetaBrainz: https://metabrainz.org/

**Comment by mateusz-bak** [comment]
Hey @felysu thanks for the suggestion! Will take a look at this source and check if it fits with the app.

**Comment by ropery** [comment]
https://isbndb.com seems large/useful too.

**Comment by jakejoh** [comment]
> Would you add Google Book's API as an option if user wants to enable?
> 
> I'm aware of the fact that this may not be privacy respecting. I think, some user would attract if it has Google, as OpenLibrary is not that good source.
> 
> https://developers.google.com/books/docs/v1/using
> 
> Edit: or as Goodreads doens't provide API anymore, I'll try to see if I can make a custom API for Goodreads

I made a quick and verrrrrrry dirty macro for macrodroid that can be triggered either by e.g. a quicktile or when opening the form for adding a book (triggered by screencontent).
It shows the first 3 results from google books, pastes title, author etc. to the respective fields and downloads a cover (small & if available larger) for manual selection.
https://pastebin.com/S5Px937q

**Comment by apobrt** [comment]
> https://isbndb.com seems large/useful too.

It seems proprietary and not free

**Comment by apobrt** [comment]
> BookBrainz is a new user-based book database with a clean interface and an active team, which could be very useful as an alternative data provider in the app.

It is still in a beta stage, under development, but why not?

> Would you add Google Book's API as an option if user wants to enable?

Usually I prefer not to use proprietary services and instead support open source ones (like OpenLibrary or BookBrainz), but maybe adding this feature as disabled by default, as an opt-in with very clear alert to the end user could be sensible and useful.

**Comment by x51mon** [comment]
I agree with @apobrt about a non-free data provider. I prefer using free and open-source one but I've personally stopped using the app because I found it to annoying having to enter almost every book manually.

**Comment by edern-appere** [comment]
I'm in the same situation : I prefer to rely on free and open-source software but if I cannot find my books half the time with open databases, it's anoying and I no longer use the app.
   
  
    
   
  
   Actually I still use an old-looking and non-free app to manage my books (https://play.google.com/store/apps/details?id=com.vgm.mylibrary) because it always finds the books i'm looking for (i think it scraps data from various sources).
   
   
   
    Le 04/09/2023 21:53 CEST, x51mon ***@***.***> a écrit :
    
   I agree with @apobrt about a non-free data provider. I prefer using free and open-source one but I've personally stopped using the app because I found it to annoying having to enter almost every book manually.

**Comment by mateusz-bak** [comment]
Thanks for all the suggestions. I won't be adding any proprietary sources at this stage.

I will take a look at BookBrainz now and see if it can be implemented.

**Comment by leoossa** [comment]
@mateusz-bak 
Hi,
I know it's been a long time since this issue was opened.
Though I think I found quite neat acceptable solution you can consider.
There's quite nice API here:
https://data.bn.org.pl/networks/bibs
https://www.bn.org.pl/aktualnosci/3345-biblioteka-narodowa-otwiera-najwieksza-polska-baze-danych-bibliograficznych.html

For example - I couldn't find this book by ISBN: 9788375611731 using current OpenLibrary search in the app, but I was able to find it via BN search.

They claim it's the biggest database in Poland:

```
Katalog Biblioteki Narodowej jest największą i najpełniejszą bazą danych bibliograficznych w Polsce – 4 678 549 rekordów bibliograficznych i 2 175 147 rekordów wzorcowych
```


and the API it's quite easy to consume.
